ios/brave-ios/Sources/Shared/Extensions/URLExtensions.swift

```diff
@@ -355,6 +355,49 @@ extension URL {
     return urlHost == "talk.brave.com"
   }
 
+  // Check if the website is a night mode blocked site
+  public var isNightModeBlockedURL: Bool {
+    // The reason we use `normalizedHost` is because we want to keep the eTLD+1
+    // IE: (search.brave.com instead of brave.com)
+    guard let urlHost = self.normalizedHost(), let registry = self.publicSuffix else {
+      return false
+    }
+
+    // Remove the `registry` so we get `search.brave` instead of `search.brave.com`
+    // We get amazon instead of amazon.co.uk
+    // mail.proton instead of mail.proton.com
+    let domainName = urlHost.dropLast(registry.count).trimmingCharacters(
+      in: CharacterSet(charactersIn: ".")
+    )
+
+    // Site domains that should NOT inject night mode
+    let siteList = [
```

Comment:
If we're re-adding this same list, lets use a `Set` instead of an array for more performant lookup

Reply:
Good idea. Will do.